### PR TITLE
Remove Starboard thread, SbEvent and most of ApplicationAndroid

### PR DESCRIPTION
**Pull request by johnxwork**
The Starboard application becomes very lightweight and we will 
likely remove it completely later.

Removed a few java->native->java calls related to lifecycle, but 
most initialization calls are kept.

b/377042903
b/375459298
b/375058047
b/375669373

### CODE REVIEW

File: cobalt/android/apk/app/src/main/java/dev/cobalt/coat/CobaltActivity.java [hunk 1]
```diff
@@ -328,9 +328,6 @@ public WebContents getActiveWebContents() {
   // TODO(b/375442742): re-enable native code.
   // private static native void nativeLowMemoryEvent();
 
-  // TODO(cobalt): make WebContent accessible in CobaltActivity or StarboardBridge.
```
Comment (andrewsavage1) [inline cobalt/android/apk/app/src/main/java/dev/cobalt/coat/CobaltActivity.java:331]: Can we add a bug number?

File: starboard/shared/starboard/application.h [hunk 2]
```diff
@@ -362,6 +362,10 @@ class Application {
     command_line_.reset(new CommandLine(argc, argv));
   }
 
+  void SetCommandLine(std::unique_ptr<CommandLine> command_line) {
```
Comment (andrewsavage1) [inline starboard/shared/starboard/application.h:365]: How is this used in this PR?

File: cobalt/android/apk/app/src/main/java/dev/cobalt/coat/CobaltActivity.java (hunk 1 above)
Comment (johnxwork) [reply cobalt/android/apk/app/src/main/java/dev/cobalt/coat/CobaltActivity.java:331]: I'm removing this comment. I haven't seen a need for ContentView from Activity yet. And WebContents is already available here.

File: starboard/shared/starboard/application.h (hunk 2 above)
Comment (johnxwork) [reply starboard/shared/starboard/application.h:365]: It's used on line 63 of starboard/android/shared/application_android.cc in this PR. You'd want to expand the file(I don't know why github is collapsing it... It's not a removal)

**Review by kaidokert — APPROVED** [review]
lgtm

**Review by y4vor — COMMENTED** [review]
Why are we removing the application? We'll have to bring it back for AOSP at some point. Maybe a better approach is to have optionally create it based on whether Starboard is used as a library or as a Platform.

**Comment by johnxwork** [comment]
> Why are we removing the application? We'll have to bring it back for AOSP at some point. Maybe a better approach is to have optionally create it based on whether Starboard is used as a library or as a Platform.

Synced offline and decided to keep the original application_android for AOSP usage. We will name it application_aosp moving forward since atv will not use SbEvents

File: starboard/android/shared/BUILD.gn [hunk 3]
```diff
@@ -323,6 +322,12 @@ static_library("starboard_platform") {
     "//starboard/shared/starboard/player/player_set_playback_rate.cc",
   ]
 
+  # TODO(cobalt, b/378971929): Turn on ApplicationAndroid when working on AOSP.
```
Comment (y4vor) [inline starboard/android/shared/BUILD.gn:325]: This should be gated with `is_starboard`.

File: starboard/android/shared/android_main.cc [hunk 4]
```diff
@@ -44,16 +44,6 @@ std::atomic_bool g_block_swapbuffers{false};
 namespace {
 
 using ::starboard::shared::starboard::CommandLine;
-typedef ::starboard::android::shared::ApplicationAndroid::AndroidCommand
```
Comment (y4vor) [inline starboard/android/shared/android_main.cc:47]: Have a separate android_main_aosp.cc that keeps all that a logic?

File: starboard/android/shared/aosp_main.cc [hunk 5]
```diff
@@ -0,0 +1,366 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <dirent.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <atomic>
+
+// #include "game-activity/GameActivity.h"
```
Comment (kaidokert) [inline starboard/android/shared/aosp_main.cc:21]: Well. With these changes all of `aosp` stuff is basically unbuilt, untested code. Do we need it, now ? Or can we add it back in when we are actually building it ( 25lts branch isn't going anywhere, we can always take it from there )

Assumptions like "IS_EVERGREEN_COMPATIBLE" and such in this file don't even make sense with AOSP context.

**Review by kaidokert — CHANGES_REQUESTED** [review]
Re-consider the AOSP aspect. 

You are adding a host of unbuilt, untested code with untracked changes effectively, with assumptions that aren't checked either at build or runtime. This doesn't help us with our immediate objectives, and the code isn't going anywhere - it's still available to copy back in ( with tests ) once we start building it in 25lts branch.

File: starboard/android/shared/aosp_main.cc (hunk 5 above)
Comment (yell0wd0g) [reply starboard/android/shared/aosp_main.cc:21]: Could we just leave a giant NOTIMPLEMENTED() in the AOSP files?

File: cobalt/android/apk/app/src/main/java/dev/cobalt/coat/StarboardBridge.java [hunk 6]
```diff
@@ -113,9 +114,9 @@ public StarboardBridge(
 
     Log.i(TAG, "StarboardBridge init.");
 
-    // Make sure the JNI stack is properly initialized first as there is
+    // Make sure the JNI stack is properly initialized first as there is a
     // race condition as soon as any of the following objects creates a new thread.
-    nativeInitialize();
+    initJNI();
```
Comment (zhongqiliang) [inline cobalt/android/apk/app/src/main/java/dev/cobalt/coat/StarboardBridge.java:119]: When CoAT stops, JNI will be detached from the jvm thread. 
In lts.25, the entry point is the Destroy() method in jni_env_ext.cc, do you still have the method called when the app stops?

File: starboard/android/shared/aosp_main.cc (hunk 5 above)
Comment (johnxwork) [reply starboard/android/shared/aosp_main.cc]: Merged aosp_main.cc and android_main.cc with #SB_IS(EVERGREEN_COMPATIBLE) macro

File: cobalt/android/apk/app/src/main/java/dev/cobalt/coat/StarboardBridge.java (hunk 6 above)
Comment (johnxwork) [reply cobalt/android/apk/app/src/main/java/dev/cobalt/coat/StarboardBridge.java:119]: Damn this is a good fine. How did we have a JNI shutdown in SbThread orz

File: starboard/android/shared/aosp_main.cc (hunk 5 above)
Comment (johnxwork) [reply starboard/android/shared/aosp_main.cc]: Removed aosp files from this PR and into its own one: https://github.com/youtube/cobalt/pull/4436/files

We can unblock this one and continue the discussion there.

**Review by kaidokert — APPROVED** [review]
Okay with this as is - please don't change it again.

**Pull request merged**